### PR TITLE
Update PureFA integration to v1.1.0

### PR DESCRIPTION
**Pull request by chrroberts-pure**
### What does this PR do?

This PR is to introduce support for our new [Pure Storage OpenMetrics Exporter](https://github.com/PureStorage-OpenConnect/pure-fa-openmetrics-exporter). It includes new metric names and dashboards for the new exporter.

### Motivation

What inspired you to submit this pull request?

The legacy Prometheus exporter was deprecated in favor of [Pure Storage OpenMetrics Exporter](https://github.com/PureStorage-OpenConnect/pure-fa-openmetrics-exporter)

### Review checklist

- [ ] PR has a [meaningful title](https://github.com/DataDog/integrations-core/blob/master/CONTRIBUTING.md#pull-request-title) or PR has the `no-changelog` label attached
- [ ] Feature or bugfix has tests
- [ ] Git history is clean
- [ ] If PR impacts documentation, docs team has been notified or an issue has been opened on the [documentation repo](https://github.com/DataDog/documentation/issues/new)

### Additional Notes

Anything else we should know when reviewing?

### CODE REVIEW

**Comment by chrroberts-pure** [comment]
@sdodsley

**Comment by chrroberts-pure** [comment]
Hi. Can I please get an update on this approval process to get this merged? I have some customers of ours who are waiting for this PR to be merged. @anawishnoff

**Comment by anawishnoff** [comment]
@chrroberts-pure Very sorry for the delay! We'll get this reviewed immediately.

File: purefa/README.md
Comment (anawishnoff) [inline purefa/README.md:106]:
```suggestion
The dashboard for the deprecated [Pure Storage Prometheus exporter][14] has been renamed to `Pure FlashArray - Overview (Legacy Exporter)`
```

File: purefa/README.md
Comment (anawishnoff) [inline purefa/README.md:108]:
```suggestion
A listing of metrics that are both shared and unique to the different exporters are listed in [metrics.py][15]. You may need to update your dashboards and/or your alerts to match the new metrics names when migrating from the [Pure Storage Prometheus exporter][14] to the [Pure Storage OpenMetrics exporter][1]. Please contact Pure Storage with the information in the Support tab if you have any questions.
```

File: purefa/README.md
Comment (anawishnoff) [inline purefa/README.md:110]: Wouldn't the migration be the other way around for this update? Prometheus -> OpenMetrics?

File: purefa/README.md
Comment (anawishnoff) [inline purefa/README.md:100]:
```suggestion
### Upgrading to new versions of this integration
```

This feels clearer to me, but feel free to re-word as needed

**Review by anawishnoff — CHANGES_REQUESTED** [review]
Comments from the product side

File: purefa/README.md
Comment (chrroberts-pure) [reply purefa/README.md:110]: Good catch. Updated.

File: purefa/README.md
Comment (chrroberts-pure) [reply purefa/README.md:108]: Good catch, i didn't consider how it's viewed in the datadog interface.

**Review by mgarabed — APPROVED** [review]
Looks good from Marketplace perspective.

File: purefa/README.md
Comment (steveny91) [inline purefa/README.md:15]: 👋 For this setup do you envision users using a containerized agent as well? The reason why I'm asking is because installing the an extras check in a containerized agent isn't really straightforward. But if the intended setup involves just having the agent in the host/node and polling metrics from containers then it should be fine.

File: purefa/README.md
Comment (alai97) [inline purefa/README.md:108]:
```suggestion
A listing of metrics that are both shared and unique to the different exporters are listed in [metrics.py][15]. You may need to update your dashboards and/or your alerts to match the new metric names when migrating from the [Pure Storage Prometheus exporter][14] to the [Pure Storage OpenMetrics exporter][1]. Please contact Pure Storage with the information in the Support tab if you have any questions.
```

File: purefa/README.md
Comment (alai97) [inline purefa/README.md:114]:
```suggestion
In future versions of the PureFA Agent Check, the metric names from the Pure Storage Prometheus exporter will be removed.
```

File: purefa/metadata.csv
Comment (alai97) [inline purefa/metadata.csv:1]: Minor nit that metric descriptions should all end with periods, but this is not required formatting.

**Review by alai97 — APPROVED** [review]
Looks good, some copy suggestions!

File: purefa/metadata.csv
Comment (chrroberts-pure) [reply purefa/metadata.csv:1]: fixed!

File: purefa/README.md
Comment (chrroberts-pure) [reply purefa/README.md:15]: I'd really like it to be that way! So I'd like to keep the option open to our customers. 

The idea would be for the datadog-agent containers running in k8s to directly query the OpenMetrics containers that we have an image for (linked in the readme) in the same cluster.

File: purefa/datadog_checks/purefa/metrics.py
Comment (iliakur) [inline purefa/datadog_checks/purefa/metrics.py:4]: If you're only defining a `name` key, you can use the shorthand:
```python
"purefa_info": "info",
```

In other words, convert all your dicts to strings

**Review by iliakur — COMMENTED** [review]
I refrain from approval because the question Steven asked should be addressed.

File: purefa/README.md
Comment (steveny91) [reply purefa/README.md:15]: Gotcha. I envision some difficulty with customers using the check with a containerized agent because the agent will need to be restarted after the check/integration installation step, which isn't really easy since the container will just get deleted and a new one will be created to take it's place. But that might be outside the scope of this PR.

**Review by steveny91 — APPROVED** [review]

File: purefa/datadog_checks/purefa/metrics.py
Comment (chrroberts-pure) [reply purefa/datadog_checks/purefa/metrics.py:4]: I would prefer to keep it this way as I was having issues in the past with default_config when using the shorthand.

File: purefa/README.md
Comment (chrroberts-pure) [reply purefa/README.md:15]: Agree, but will keep in mind and I'll probably reach out if any customers specifically require it. Also there's also other methods of ingestion that we are working on that could even bypass the need for this agent integration. (PureFA OTel Receiver)

**Review by anawishnoff — APPROVED** [review]

**Pull request merged**
